streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -755,16 +753,25 @@ private TaskAssignor createTaskAssignor(final boolean lagComputationSuccessful)
     private boolean populateClientStatesMap(final Map<UUID, ClientState> clientStates,
                                             final Map<UUID, ClientMetadata> clientMetadataMap,
                                             final Map<TopicPartition, TaskId> taskForPartition,
-                                            final Map<TaskId, Set<TopicPartition>> changelogsByStatefulTask) {
+                                            final Map<TaskId, Set<TopicPartition>> changelogsByStatefulTask,
+                                            final Set<String> newlyCreatedChangelogs) {
         boolean fetchEndOffsetsSuccessful;
         Map<TaskId, Long> allTaskEndOffsetSums;
         try {
-            final Collection<TopicPartition> allChangelogPartitions = changelogsByStatefulTask.values().stream()
-                                                                          .flatMap(Collection::stream)
-                                                                          .collect(Collectors.toList());
+            final Collection<TopicPartition> allExistingChangelogPartitions =
```

Comment:
I like immutability. Should we call this `allExpetcedChangelogPartitions` and introduce `allExistingChangelogPartitions = allExpetcedChangelogPartitions - newlyCreatedChangelogs`

Reply:
Fair enough